### PR TITLE
Esy: add libs: memtrace, benchmarks and landmark in esy

### PR DESCRIPTION
**Pull request by lykimq**
Add libs for benchmarks in Esy.

### CODE REVIEW

**Review by aguillon — APPROVED** [review]
Will merge if @ulrikstrid/@callistonianembrace approve.

**Comment by ulrikstrid** [comment]
@lykimq What I usually do when I have conflicts in `esy.lock` is to rebase on `main` then just pick whatever of the 2 alternatives and before commiting I do `esy install` to get a fresh lockfile and commit that in the rebase.

File: esy.json [hunk 1]
```diff
@@ -44,7 +44,10 @@
     "@opam/data-encoding": "*",
     "@opam/prometheus-dream": "*",
     "@opam/conf-libssl": "3",
-    "@opam/json-logs-reporter": "marigold-dev/json-logs-reporter:json-logs-reporter.opam#ff835c33b45eb619a79d42dbce0eb88d16bc27db"
+    "@opam/json-logs-reporter": "marigold-dev/json-logs-reporter:json-logs-reporter.opam#ff835c33b45eb619a79d42dbce0eb88d16bc27db",  
+    "@opam/memtrace": "*", 
+    "@opam/landmarks": "*", 
+    "@opam/benchmark": "*"
```
Comment (ulrikstrid) [inline esy.json:50]: I'm surprised it's this easy now, what did we change? :joy:

**Comment by lykimq** [comment]
> @lykimq What I usually do when I have conflicts in `esy.lock` is to rebase on `main` then just pick whatever of the 2 alternatives and before commiting I do `esy install` to get a fresh lockfile and commit that in the rebase.

Thanks, I will do it.

File: esy.json (hunk 1 above)
Comment (ulrikstrid) [reply esy.json:50]: Aha, `core_bench` is the problematic dependency, I was just confused

**Pull request merged**